pkg/csi/service/vanilla/controller.go

```diff
@@ -1077,15 +1035,15 @@ func (c *controller) ListVolumes(ctx context.Context, req *csi.ListVolumesReques
 	ctx = logger.NewContextWithLogger(ctx)
 	log := logger.GetLogger(ctx)
 	log.Infof("ListVolumes: called with args %+v", *req)
-	return nil, status.Error(codes.Unimplemented, "")
+	return nil, logger.LogNewErrorCode(log, codes.Unimplemented, "")
```

Comment:
Why do we need to change this? We are not logging anything explicitly here right?

Reply:
Should we explicitly log for errors? Why not add a more user readable error message?

Reply:
The message in this case is an empty string right, so I did not see the need to change this line.

Reply:
The difference between status.Error and logger.LogNewErrorCode is that the latter adds a log
message to the logger. I suppose that you are arguing that there is no need to log the error,
if it is an "Unimplemented" error.

I argue that we should log all errors. So, we should always use logger.LogNewErrorCode.
My second argument is that the empty message is not helpful for debugging. It would be better
to have an explicit error message, which could be done in another change by the code owner.

Did I misunderstand you?

Reply:
> I suppose that you are arguing that there is no need to log the error,
if it is an "Unimplemented" error.
No I did not mean that. We should log the error wherever necessary but in this case, you are not logging anything. You are sending an empty string to the logger. Given that, I wanted to know what is the difference between using a direct `status.Error` vs an `logger.LogNewErrorCode` with an empty log message.

Reply:
If you plan to include an error message instead of an empty string in a separate commit, go ahead. I will approve this PR

Reply:
Thanks, Shalini.

How about we just add the method name as the message?

Reply:
Done.

Reply:
Sounds good. You could also add a simple message like `<MethodName> is not yet implemented`